src/ringbuffer.c

```diff
@@ -0,0 +1,75 @@
+#include "ringbuffer.h"
+
+/**
+ * @file
+ * Implementation of ring buffer functions.
+ */
+
+void ring_buffer_init(ring_buffer_t *buffer) {
+  buffer->tail_index = 0;
+  buffer->head_index = 0;
+}
+
+void ring_buffer_queue(ring_buffer_t *buffer, char data) {
+  /* Is buffer full? */
+  if(ring_buffer_is_full(buffer)) {
```

Comment:
reducing indent level (cyclomatic complexity: https://en.wikipedia.org/wiki/Cyclomatic_complexity) by inverting the statement:

```c
if (!ring_buffer_is_full(buffer))
    return;
buffer->tail_index = ((buffer->tail_index + 1) & RING_BUFFER_MASK);
```

Reply:
There are statements after this . So immediately returning doens't seem correct.

```C
void ring_buffer_queue(ring_buffer_t *buffer, char data) {
  /* Is buffer full? */
  if(ring_buffer_is_full(buffer)) {
    /* Is going to overwrite the oldest byte */
    /* Increase tail index */
    buffer->tail_index = ((buffer->tail_index + 1) & RING_BUFFER_MASK);
  }

  /* Place data in buffer */
  buffer->buffer[buffer->head_index] = data;
  buffer->head_index = ((buffer->head_index + 1) & RING_BUFFER_MASK);
}
```